src/decoder.rs

```diff
@@ -176,6 +221,10 @@ pub fn decode_regular(rsm: RawSourceMap) -> Result<SourceMap> {
                 }
             }
 
+            if rmi.trailing_zeros() != 0 && line_index == rmi.trailing_zeros() {
+                range_tokens.push(tokens.len() as u32)
+            }
```

Comment:
I don't understand what the number of trailing zeros in `rmi` has to do with where the range mappings are. Also, this still implies that there is at most one range mapping per line (at most one `line_index` can be equal to `rmi.trailing_zeros`).